### PR TITLE
Wip/reportsperformance

### PR DESCRIPTION
**Pull request by sulkaharo**
Rendering two weeks of graphs on my laptop drops from over a minute to around 9 seconds with these. Using binary search for temps, deduping data, caching latest temp result for huge gain in reporting speed & loading less temp basal data. Note the new loadash version.

### CODE REVIEW

File: lib/profilefunctions.js [hunk 1]
```diff
@@ -158,8 +160,21 @@ function init(profileData) {
...
     profile.tempbasaltreatments = tempbasaltreatments || [];
+
+	// dedupe temp basal events    
+    profile.tempbasaltreatments = _.uniqBy(profile.tempbasaltreatments, 'mills');
+    
+    _.each(profile.tempbasaltreatments, function addDuration(t) {
+       t.endmills = t.mills + times.mins(t.duration || 0).msecs;
+    });
+    
+    profile.tempbasaltreatments.sort(function (a, b) {
```
Comment (jasoncalabrese) [inline lib/profilefunctions.js]: Codacy Issue found: [Missing function expression name.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2437405937/issues/source?bid=2452379&fileBranchId=3244148#l174)

File: lib/profilefunctions.js [hunk 2]
```diff
@@ -158,8 +160,21 @@ function init(profileData) {
...
 
   profile.updateTreatments = function updateTreatments (profiletreatments, tempbasaltreatments, combobolustreatments) {
+  
     profile.profiletreatments = profiletreatments || [];
     profile.tempbasaltreatments = tempbasaltreatments || [];
+
+	// dedupe temp basal events    
+    profile.tempbasaltreatments = _.uniqBy(profile.tempbasaltreatments, 'mills');
+    
+    _.each(profile.tempbasaltreatments, function addDuration(t) {
```
Comment (jasoncalabrese) [inline lib/profilefunctions.js]: Codacy Issue found: [Missing space before function parentheses.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/2437405937/issues/source?bid=2452379&fileBranchId=3244148#l170)

**Comment by MilosKozak** [comment]
if this is performance improvement voting for merge regardless we do with loading of temps

**Comment by jasoncalabrese** [comment]
I wasn't sure how this lined up with the other PR, but looks good to me

**Pull request merged**